rust/connlib/tunnel/src/client.rs

```diff
@@ -281,37 +281,61 @@ fn send_dns_answer(
     }
 }
 
+/// A SANS-IO implementation of a client's functionality.
+///
+/// Internally, this composes a [`snownet::ClientNode`] with firezone's policy engine around resources.
+/// Clients differ from gateways in that they also implement a DNS resolver for DNS resources.
+/// They also initiate connections to gateways based on packets sent to resources.
 pub struct ClientState {
+    /// The [`snownet::ClientNode`].
+    ///
+    /// Manages wireguard tunnels to gateways.
+    node: ClientNode<GatewayId, RelayId>,
+    /// All peers gateways we are connected to and the associated, connection-specific state.
```

Comment:
Does that means the gateways of all peers we are connected to ("all peers' gateways") or all peers that are gateways? ("all peer gateways")

Reply:
Typo!

The word "peers" shouldn't be there :)

```suggestion
    /// All gateways we are connected to and the associated, connection-specific state.
```